segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/BookKeeperConfig.java

```diff
@@ -35,6 +35,7 @@
     public static final Property<Integer> BK_ACK_QUORUM_SIZE = Property.named("bkAckQuorumSize", 3);
     public static final Property<Integer> BK_WRITE_QUORUM_SIZE = Property.named("bkWriteQuorumSize", 3);
     public static final Property<Integer> BK_WRITE_TIMEOUT = Property.named("bkWriteTimeoutMillis", 5000);
+    public static final Property<Integer> BK_READ_TIMEOUT = Property.named("readTimeoutMillis", 5000);
```

Comment:
This change makes sense given the description in the associated issue #2235, but I'm wondering the following. I'm not sure it would be trivial for an operator to decide when to change the default or even to what value. When BK is used in an embedded manner like in Pravega, it is difficult to set this timeout value (or even the write timeout now that I think about it). 

Assuming that we retry the reads in the case of timeout, I'd say that it is the best we can do (very likely that we do it already, but I can't remember right now). Otherwise, we will be making the system difficult to operate in production.

Reply:
Consider this case (which is not the only one, but the one that I ran into): I connected to the remote cluster to diagnose the issue. Since it had an insane ping from my computer (hundreds of millis), I had to go to `BookKeeperLogFactory` to manually enter some astronomical value for the read timeout, otherwise recovery wouldn't happen. I prefer setting a quick config value over to figuring out why this wouldn't work.

In the majority of cases, we just leave the defaults (5s). But in some rare cases, having a harmless config will be of extreme help.

Reply:
> I had to go to BookKeeperLogFactory to manually enter some astronomical value for the read timeout, otherwise recovery wouldn't happen. I prefer setting a quick config value over to figuring out why this wouldn't work.

Right, here you are arguing about the value of a parameter over code changes. In both cases, we need to bring the server down to make the change effective, yes?

To be clear, I'm not against adding the parameter, I'm concerned that this is not going to solve the underlying problem, which is the one of preventing slow reads (not sure why reads would take long unless they are very large) from making a segment container unavailable.

Reply:
Sure. But it's a huge difference in bringing down the server, recompiling, redeploying and restarting and bringing down the server, changing a config value and restarting.

This is a trivial change. I don't understand your opposition to it. If you have a problem with it, then why not have a problem with the existing write timeout?

Reply:
> But it's a huge difference in bringing down the server, recompiling, redeploying and restarting and bringing down the server, changing a config value and restarting.

Agreed, although ideally we shouldn't have to bring down a server.

> This is a trivial change.

Agreed.

>  I don't understand your opposition to it.

This is what I said, perhaps it wasn't clear:

```
 I'm not against adding the parameter, I'm concerned that this is not going to solve the 
underlying problem, which is the one of preventing slow reads (not sure why reads would take 
long unless they are very large) from making a segment container unavailable.
```

> If you have a problem with it, then why not have a problem with the existing write timeout?

I actually did mention it above:

```
When BK is used in an embedded manner like in Pravega, it is difficult to set this timeout value (or even the write timeout now that I think about it).
```